Resumos redes (#158)
OpenDevUFCG/Tamburetei: redes/resumos/camadaFisica.md

## Hist�rico

### Evolu��o na Comunica��o e Processamento

A necessidade de comunica��o, assim como o instinto de sobreviv�ncia, � inerente aos seres humanos desde seu surgimento. Das pinturas rupestres, pombo-correio, at� a populariza��o da Internet, s�o mais de 30.000 anos de hist�ria evolutiva dos meios de comunica��o.

> *�Cada um dos tr�s s�culos anteriores foi dominado por um a �nica tecnologia. O S�culo XVIII foi a �poca dos grandes sistemas mec�nicos que acompanharam a Revolu��o Industrial. O S�culo XIX foi a era das m�quinas a vapor. As principais conquistas tecnol�gicas do S�culo XX se deram no campo da aquisi��o, do processamento e da distribui��o de informa��es. Entre outros desenvolvimentos, vimos a instala��o das redes de telefonia em escala mundial, a inven��o do r�dio e da televis�o, o nascimento e o crescimento sem precedentes da ind�stria de inform�tica e o lan�amento dos sat�lites de comunica��o.�* (TANENBAUM)

No caso espec�fico do surgimento da Internet, o contexto hist�rico � o seguinte - Guerra Fria. O mundo ainda est� se recuperando dos destro�os remanescentes da segunda guerra mundial e o clima de paz � mais falso que a amizade entre USA e a antiga URSS. 

> A comunica��o entre as bases militares tornou-se algo crucial e assunto de primeira ordem - quem tem mais informa��o consequentemente ganha a guerra. Assim, fica evidente a necessidade da cria��o de algo novo, uma nova forma de inter-comunicar essas bases� a internet! Ou pelo menos os primeiros passos dela. A internet surge ent�o, bem como outras incont�veis tecnologias, de forma n�o-exatamente-intencional. 

Reviewer: ```suggestion
A comunicação entre as bases militares tornou-se algo crucial e assunto de primeira ordem - quem tem mais informação consequentemente ganha a guerra. Assim, fica evidente a necessidade da criação de algo novo, uma nova forma de inter-comunicar essas bases… a internet! Ou pelo menos os primeiros passos dela. A internet surge então, bem como outras incontáveis tecnologias, de forma _não-exatamente-intencional_. 
```

Ítalico no "não-exatamente-internacional". 

**obs: vc pode aceitar a sugestões e o github já commita diretamente**

Em 1969, a empresa de ARPA (Advanced Research and Projects Agency), resolveu criar a famosa e saudosa (n�o pelos sovi�ticos, claro) ARPANet. Ela tinha como principal objetivo interligar as bases militares e os departamentos de pesquisas do governo americano. O projeto nasce com a guerra, mas chegado o seu fim e o sucesso dos americanos, qual o futuro da ARPANet? Depois de ser amplamente utilizada durante os embates, os militares decidiram �passar a guarda� para os cientistas, depois para as grandes universidades e assim foi sendo aprimorada e difundida (dando um grande pulo hist�rico). 

A tabela abaixo explica um pouco da evolu��o dos computadores e processamento.

**Gera��o** | **Computadores** | **Processamento** 
:---: | :---: | :---:

> **Primeira Gera��o (1940-1959)** | Exemplo: ENIAC. V�lvulas e circuitos. Ocupava o espa�o de um pr�dio. | Processamento batch: ocorre atrav�s de um lote de tarefas enfileiradas. Feito manualmente por uma equipe.

Reviewer: ```suggestion
**Primeira Geração (1940-1959)** | Exemplo: *ENIAC*. Válvulas e circuitos. Ocupava o espaço de um prédio. | *Processamento batch*: ocorre através de um lote de tarefas enfileiradas. Feito manualmente por uma equipe.
```

> **Segunda Gera��o (1959-1965)** | Exemplo: IBM 7094. V�lvulas foram substitu�das por transistores, o que diminuiu o tamanho dos computadores, permitindo o in�cio de seu uso comercial. | Os primeiros terminais interativos surgiram, onde os usu�rios poderiam acessar o computador central pelas linhas de comunica��o.

Reviewer: ```suggestion
**Segunda Geração (1959-1965)** | Exemplo: **IBM 7094**. Válvulas foram substituídas por **transistores**, o que diminuiu o tamanho dos computadores, permitindo o início de seu **uso comercial**. | Os primeiros **terminais interativos** surgiram, onde os usuários poderiam acessar o **computador central** pelas **linhas de comunicação**.
```

> **Terceira Gera��o (1965-1975)** | A populariza��o dos circuitos integrados contribuiu para, al�m de baratear o custo dos computadores, diminuir ainda mais seu tamanho, sendo agora poss�vel a ideia de computadores pessoais. | Come�o da descentraliza��o e individualiza��o no processamento e servi�os, a partir da d�cada de 70.

Reviewer: ```suggestion
**Terceira Geração (1965-1975)** | A popularização dos **circuitos integrados** contribuiu para, além de baratear o custo dos computadores, diminuir ainda mais seu tamanho, sendo agora possível a ideia de computadores pessoais. | Começo da **descentralização** e **individualização** no processamento e serviços, a partir da década de 70.
```

> **Quarta Gera��o (1975-hoje)** | Populariza��o e crescente dos computadores pessoais. | Sistemas distribu�dos e redes locais.

Reviewer: ```suggestion
**Quarta Geração (1975-hoje)** | Popularização e crescente dos **computadores pessoais**. | Sistemas distribuídos e redes locais.
```

## Defini��es e Conceitos

### O que � Internet?

> Kurose-Ross, no livro Redes de Computadores e a Internet, fala que podemos definir a Internet como �uma infraestrutura de rede que prov� servi�os para aplica��es distribu�das. � Claro que essa defini��o � bastante rasa, mas por meio dela � poss�vel entender a ponta do iceberg.

Reviewer: ```suggestion
Kurose-Ross, no livro Redes de Computadores e a Internet, fala que podemos definir a Internet como “*uma infraestrutura de rede que provê serviços para aplicações distribuídas.* ” Claro que essa definição é bastante rasa, mas por meio dela é possível entender a ponta do iceberg.
```

> Na Internet, existem dois conceitos essenciais para sua forma de comunica��o: hospedeiros e sistemas finais. Um sistema final s�o todos os computadores interligados na periferia da Internet. Hospedeiros (ou hosts) � um sistema final que, como o nome j� adianta, hospeda uma aplica��o/servi�o.

Reviewer: ```suggestion
Na Internet, existem dois conceitos essenciais para sua forma de comunicação: **hospedeiros e sistemas finais**. Um sistema final são todos os computadores interligados na periferia da Internet. **Hospedeiros (ou hosts)** é um sistema final que, como o nome já adianta, hospeda uma aplicação/serviço.
```

> Segundo Kurose-Ross (vai ter muito ele aqui mesmo), �Sistemas finais n�o s�o interligados diretamente por um �nico enlace de comunica��o. Em vez disso, s�o interconectados indiretamente por equipamentos intermedi�rios de comuta��o conhecidos como comutadores de pacotes.� Um exemplo de comutador de pacotes s�o roteadores, que encaminham pacotes (bloco de dados) para seus destinos finais.

Reviewer: ```suggestion
Segundo Kurose-Ross (vai ter muito ele aqui mesmo), “Sistemas finais não são interligados diretamente por um único enlace de comunicação. Em vez disso, são interconectados indiretamente por equipamentos intermediários de comutação conhecidos como **comutadores de pacotes**.” Um exemplo de comutador de pacotes são roteadores, que encaminham pacotes (bloco de dados) para seus destinos finais.
```

Os sistemas finais s�o conectados entre si por enlaces de comunica��o, como por exemplo: fibra �ptica (cabeada) e ondas de r�dio (sem fio), e acessam a Internet atrav�s de ISPs (Internet Service Providers). Cada ISP � uma rede de comutadores de pacotes e enlaces de comunica��o.

### O que s�o RFCs e a IETF?

> IETF - Internet Engineering Task Force. Trata-se de um grupo respons�vel por desenvolver documentos padr�es para tecnologias que cercam a Internet, os RFCs.

Reviewer: ```suggestion
**IETF** - **Internet Engineering Task Force**. Trata-se de um grupo responsável por desenvolver documentos padrões para tecnologias que cercam a Internet, os RFCs.
```

> RFC - Request  for Comments. Elas come�aram como solicita��es gerais em coment�rios para resolver problemas que a Internet enfrentava no in�cio. Tendem a ser detalhados e t�cnicos e existem mais de 7.000.

Reviewer: ```suggestion
**RFC** - **Request  for Comments**. Elas começaram como solicitações gerais em comentários para resolver problemas que a Internet enfrentava no início. Tendem a ser detalhados e técnicos e existem mais de 7.000.
```

### Conceito de Redes de Computadores

Segundo Tanenbaum, Redes de Computadores podem ser definidas como **�um conjunto de computadores aut�nomos interconectados por uma �nica tecnologia.�**

> *Dois computadores est�o interconectados quando podem trocar informa��es. A conex�o n�o precisa ser feita por um fio de cobre; tamb�m podem ser usadas fibras �pticas, micro-ondas, ondas de infravermelho e sat�lite de comunica��es. Existem redes em muitos tamanhos, modelos e formas(...). Embora possa parecer estranho para algumas pessoas, nem a Internet nem a World Wide Web � uma rede de computadores. (...) A resposta simples � que a Internet n�o � uma �nica rede, mas uma rede de redes, e a Web � um sistema distribu�do que funciona na Internet.�*

### O que � Protocolo?

> Protocolo, no �mbito de redes de computadores, pode ser definido como um conjunto de padr�es e regras utilizados a fim de possibilitar a comunica��o entre dispositivos. Eles controlam o envio e a recep��o de mensagens.

Reviewer: ```suggestion
**Protocolo**, no âmbito de redes de computadores, pode ser definido como um conjunto de padrões e regras utilizados a fim de possibilitar a comunicação entre dispositivos. Eles controlam o envio e a recepção de mensagens.
```

> Existem diversos tipos de protocolos, cada um focado em um aspecto e funcionalidade diferente. Por exemplo, o protocolo IP (Internet Protocol), que especifica o formato dos pacotes que s�o enviados e recebidos entre roteadores e sistemas finais; o SMTP (Simple Mail Transfer Protocol) para envio de e-mails e tantos outros.

Reviewer: ```suggestion
Existem diversos tipos de protocolos, cada um focado em um aspecto e funcionalidade diferente. Por exemplo, o protocolo IP (Internet Protocol), que especifica o formato dos pacotes que são enviados e recebidos entre roteadores e sistemas finais; o **SMTP** (Simple Mail Transfer Protocol) para envio de e-mails e tantos outros.
```
